libs/article-parser/src/lib/parser/ebi-parser.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  ParsedArticle,
+  Parser,
+  ParsedArticleParagraph,
+  EbiParserOptions,
+} from '@foodmedicine/interfaces';
+import * as cheerio from 'cheerio';
+
+/**
+ * A parser for https://www.ebi.ac.uk/europepmc/webservices/rest/
+ */
+export const EbiParser: Parser<ParsedArticle> = {
+  parserF: async (xml: string, opts?: EbiParserOptions) => {
+    if (
+      !opts ||
+      !opts.parsedArticleHead ||
+      !opts.impacted ||
+      !opts.recommendation
+    )
+      throw 'Please add in the parsed head through the options';
+    const $ = cheerio.load(xml);
+    const paragraphTexts: string[] = $('p')
+      .map((i, el) => $(el).text())
+      .get();
+    const paragraphsProms: Promise<
+      ParsedArticleParagraph
+    >[] = paragraphTexts.map((paragraphText) =>
+      opts.getCorrelationScore(
+        paragraphText,
+        opts.impacted,
+        opts.recommendation
+      )
+    );
+    const paragraphs = await Promise.all(paragraphsProms);
```

Comment:
I think I mentioned this before but not sure why getCorrelationScore is an async function, seems like you should be able to do it without the promise here given that you are just running calculations over numbers and text you've already fetched

Reply:
getCorrelationScore is an async function because it may include an async fetch function or an async function to get distance between words

Reply:
The way it is set up right now, it looks like getCorrelationScore is set to getParagraphCorrelationScore, which doesn't seem to make any async calls. Is there something I'm missing there?

Reply:
Oh, yeah. I think I had some async calls in there, but then removed them. Overall though, I was thinking that it would make sense to keep it as I plan to add additional NLP which would be making async calls

Reply:
Generally, we do try to keep code checked in accounting for what is checked in rather than what will be. So keeping functions as sync instead of async unless there is already some async logic there. If you're sure the next logic will be async, I'm fine with you keeping as is, but keep in mind for future that it's generally best practice to account for what is being checked in rather than what will be :)

Reply:
Ok sure, I removed the async though and made it a normal funciton